### DISCUSSION TITLE
General question about memory use estimates on TTS models

### DISCUSSION THREAD

**Discussion by jreus** [coqui-ai/TTS#942] [General Q&A]
Hi everyone, I was just curious if there were any ways to estimate the GPU memory necessary to run a training task with a coqui model? 
I've tried running the vanilla glowTTS example on my laptop with a NVIDIA GTX 1650 with 4GB of ram, but am getting `CUDA out of memory` errors*. Do any of the more experienced coqui users think this sounds unreasonable? Or have any idea what kind of GPU resources I should aim for to run this training job?

*To be precise: `RuntimeError: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 3.82 GiB total capacity; 1.73 GiB already allocated; 14.56 MiB free; 1.79 GiB reserved in total by PyTorch`

**Comment by erogol** [comment]
For the training, the max usage occurs when the model runs on the batch that has all the longest sequences. 

It is in general the last batch of the training as the data loader sorts the instances by sequence length at the beginning of an epoch. 

But currently, there is no explicit way to estimate the GPU mem usage, unfortunately. 

Regardless 4GB is really constrained to train models.

**Reply by jstaerk** [reply]
No estimate but if one uses recipes/ljspeech/download_ljspeech.sh a tacotron2-DDC training will apparently be killed by the OOM killer around ~8h on CPUs with 32GB Ram (64GB seem to suffice), even if you tell it to kill other processes first (oom_adjust). If you have a GPU with 16GB VRAM already 30GB Ram seem to suffice.
